File: Programmers/타겟 넘버.java
//https://programmers.co.kr/learn/courses/30/lessons/43165

class Solution {
    public int solution(int[] numbers, int target) {
        int answer = 0;
        answer = dfs(0,0,numbers,target);
        return answer;
    }
    
    public int dfs(int cnt, int now, int[] numbers, int target){
        if(cnt == numbers.length){
            if(now == target) return 1;
            return 0;
        }
        return dfs(cnt+1, now+numbers[cnt], numbers, target) + dfs(cnt+1, now-numbers[cnt], numbers, target);
    }
}

File: Programmers/타겟 넘버.cpp
//https://programmers.co.kr/learn/courses/30/lessons/43165

#include <string>
#include <vector>

using namespace std;

void dfs(int cnt, int now, vector<int> &num, int target, int& answer){
    if(cnt == num.size()){
        if(now == target) answer++;
        return;
    }
    dfs(cnt+1, now+num[cnt], num, target, answer);
    dfs(cnt+1, now-num[cnt], num, target, answer);
}

int solution(vector<int> numbers, int target) {
    int answer = 0;
    dfs(0,0,numbers,target,answer);
    return answer;
}